lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java

```diff
@@ -416,6 +424,35 @@ public Optional<Location> createTemporaryDirectory(Location targetPath, String t
         return Optional.of(temporary);
     }
 
+    @Override
+    public Optional<UriLocation> preSignedUri(Location location, Duration ttl)
+            throws IOException
+    {
+        if (azureAuth instanceof AzureAuthOauth) {
```

Comment:
add `supportsPreSignedUrls` method instead

Reply:
I meant in interface :)

Reply:
oh no, that doesn't make sense imo

Reply:
Why? :)

Reply:
Because the TrinoFileSystem should have an interface as simple as possible. Adding two methods for one purpose seems like an overkill to me

Reply:
I meant AzureAuth interface